jena-base/src/main/java/org/apache/jena/atlas/io/IO.java

```diff
@@ -441,13 +441,50 @@ public static String readWholeFileAsUTF8(InputStream in) {
         }
     }
 
+    /** Read a given number of characters from a stream as UTF-8.
+     * Append a given marker such as "..." if more data remains in the stream.
+     * Closing the stream is the caller's responsibility.
+     *
+     * @param in           InputStream to be read
+     * @param maxWidth     Maximum number of bytes (rather than characters) to read from the input stream
+     * @param abbrevMarker A string to append to the result if the input stream was not fully consumed
+     * @return      String
+     */
+    public static String abbreviateAsUTF8(InputStream in, int maxWidth, String abbrevMarker) throws IOException {
+        return abbreviate(in, StandardCharsets.UTF_8, maxWidth, abbrevMarker);
```

Comment:
Use `IO.asUT8` here.

Reply:
I removed the UTF8 specific methods because the charset is now always passed as an argument anyway.